graspologic/pipeline/embed/adjacency_spectral_embedding.py

```diff
@@ -0,0 +1,209 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+from typing import Optional, Union
+import warnings
+
+import networkx as nx
+import numpy as np
+
+from . import __SVD_SOLVER_TYPES  # from the module init
+from ._elbow import _index_of_elbow
+from .embeddings import Embeddings
+from ..preconditions import (
+    check_argument,
+    check_argument_types,
+    check_optional_argument_types,
+    is_real_weighted,
+)
+from graspologic.embed import AdjacencySpectralEmbed
+from graspologic.utils import is_fully_connected, pass_to_ranks
+
+
+def adjacency_spectral_embedding(
+    graph: Union[nx.Graph, nx.DiGraph],
+    dimensions: int = 100,
+    elbow_cut: Optional[int] = None,
+    svd_solver_algorithm: str = "randomized",
+    svd_solver_iterations: int = 5,
+    svd_seed: Optional[int] = None,
+    weight_attribute: str = "weight",
+) -> Embeddings:
+    """
+    Given a weighted directed or undirected networkx graph (*not* multigraph),
+    generate an Embeddings object.
+
+    Adjacency spectral embeddings are extremely egocentric.
```

Comment:
what does egocentric mean here?

Reply:
it's a great question - I stole it from J-Lar.  He often describes ASE as "extremely egocentric" and LSE as "core periphery" but I couldn't tell you exactly what he means by that.  So we'll summon him with @JonathanLarson and see if he can shed some light on it.  I'm ok with removing it if we can't get a good justification.

Reply:
i guess my point was mainly that it seems like something we should define or have another term for in the spirit of making it developer friendly. I think is referring to something along the lines of a focus on local neighborhood of a node?

Reply:
I agree with you, which is why I'm trying to get Jon to come in here and give us more details on why he finds that a good descriptor and exactly what it means.  I have my understanding based on colloquial meanings of ego-centric, but I don't want to say they're right.